src/Sylius/Bundle/CoreBundle/spec/Applicator/ProductCatalogPromotionApplicatorSpec.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\CoreBundle\Applicator;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\CoreBundle\Applicator\ProductCatalogPromotionApplicatorInterface;
+use Sylius\Component\Core\Model\ChannelPricingInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+final class ProductCatalogPromotionApplicatorSpec extends ObjectBehavior
+{
+    function it_implements_product_catalog_promotion_applicator_interface(): void
+    {
+        $this->shouldImplement(ProductCatalogPromotionApplicatorInterface::class);
+    }
+
+    function it_applies_percentage_discount_on_all_products_variants(
```

Comment:
`on all` or `for all` that is the question 🤔

Reply:
both works, imo `on all` is sligthly better, as promotion is in fact applied **on** the product price